### ISSUE TITLE
nested-structs does not detect a slice of structs

### ISSUE THREAD

**Issue by powerman** [mgechev/revive#664]
**Describe the bug**
Rule `nested-structs` detects this:
```go
type Good struct {
  Field struct{}
}
```
but not this:
```go
type Bad struct {
  Field []struct{}
}
```

**To Reproduce**
1. I updated revive `go get -u github.com/mgechev/revive`
2. I run it with the following flags & configuration file:

```shell
$ cat bug.go
package bug

type Good struct {
	Field struct{}
}

type Bad struct {
	Field []struct{}
}
$ revive -config <(echo '[rule.nested-structs]') bug.go
bug.go:4:2: no nested structs are allowed
```

**Expected behavior**
```
$ revive -config <(echo '[rule.nested-structs]') bug.go
bug.go:4:2: no nested structs are allowed
bug.go:8:2: no nested structs are allowed
```

**Desktop (please complete the following information):**
 - OS: Gentoo Linux
 - Go: 1.18

**Comment by chavacava** [comment]
Hi @powerman, thanks for filling the issue.
I've merged the fix. #665
